### PR TITLE
feat(skill-quality): add the shared listing-budget report, fix check 2's joiner

### LINKED ISSUE #1404: skill-quality: check-skill.sh has no shared listing-budget check, plus three smaller gate defects
## Why this exists

The skill-listing measurement behind #1271 (evidence PR #1286, closed — its diff stays readable)
audited what `skill-quality:check` already enforces before proposing anything new. It found four
defects in the incumbent gate. They are recorded here because #1271 is being re-scoped to a decision
about which levers to pull, and these are code fixes that stand on their own regardless of how that
decision lands.

## 1. There is no check on the shared listing budget at all — the important one

`plugins/skill-quality/scripts/check-skill.sh:219-234` (check 2) is a **per-skill** cap: it guards
the 1,536-character per-entry truncation limit. The skill's own description advertises a
listing-budget cap, which is a different limit — the shared budget across every listed skill,
documented as scaling with 1% of the model's context window.

Nothing checks that shared budget. **That is how the listing came to overflow by a large multiple
with no gate reporting it.** The measured per-entry maximum in this tree is 1,161 characters against
a 1,536 cap, so check 2 has never fired and never would; meanwhile the aggregate is multiples over.

A check here has to decide what to do about a machine-dependent budget (it scales with context
window, and the drop order is decayed local usage), so the honest form is probably a reported
aggregate with a repo-level ceiling rather than a pass/fail on live drop behavior.

## 2. Check 2 omits the entry joiner

`check-skill.sh:227` computes `DESC_LEN + WTU_LEN`. The actual entry text the harness builds is
`description` + `" - "` + `when_to_use` — a literal three-character joiner, recovered from the
shipped binary rather than inferred. The check under-counts by 3 whenever `when_to_use` is populated.

Not currently binding, and it will not become binding at present lengths. It is wrong in exactly the
direction the listing work is about, and it is a two-character fix.

## 3. Check 12 penalizes the authoring model `when_to_use` exists to enable

`check-skill.sh:440-448` warns unless the literal text `use when` appears in `description` or
`when_to_use`. A skill that puts bare quoted trigger phrases in `when_to_use` with no prose marker
warns — measured at 15 of 15 on a prepared rewrite block.

Two fixes, and they are not equivalent:

- Keep a literal `Use when:` prefix inside `when_to_use`. No code change, but it costs roughly ten
  characters per skill on top of the joiner's three — across 130 model-invocable skills, about
  +1,700 characters, which is more than the entire recoverable mass the measurement found.
- **Amend check 12 to accept a populated, single-quoted `when_to_use` as satisfying the
  trigger-spec requirement.** This is the better fix: the warning text ("a description is a trigger
  spec, not a summary") encodes the pre-`when_to_use` authoring model.

The second edits `check-skill.sh`, so it **sequences behind #1096**, which is already editing that
file and claims check 21.

## 4. `skillListingBudgetFraction` is not at its documented default on at least one machine

The measurement found the live value is `0.03`, not the documented `0.01`. Any proposal that reasons
from the default is reasoning from the wrong number on that machine. This is a consumer setting
rather than a repo artifact, so it belongs here as a caveat on any check built for item 1: **a gate
that assumes `0.01` will disagree with `/doctor` on a machine that sets it otherwise.**

## One discrepancy worth resolving before building item 1

Two figures for the same budget are on #1271 and they differ by 4×:

- Derived from the shipped binary: the budget is `contextTokens × 4 × 0.01` **characters** — 40,000
  at 1M context, 8,000 at 200k.
- Read directly from the docs' "1% of the model's context window": 10,000 at 1M, 2,000 at 200k.

The difference is whether the 1% applies to tokens (then multiplied to characters) or to characters
directly. It changes the reported overflow multiple from roughly 2× to roughly 8× at 1M context.
**Settle this against the binary or against `/doctor`'s reported listing cost before any check
asserts a number**, since `/doctor` already reports the listing's context cost and its biggest
contributors, and overflow writes a warning to the debug log under `--debug`.

## Related

Refs #1271 (the listing-budget issue these were found under, now `status: needs-decision`).
Refs #1096 (owns `check-skill.sh` today; item 3 sequences behind it).
Refs #1286 (closed evidence PR — the full measurement, per-plugin tables, and re-verification record
live in its diff).

### PR DESCRIPTION
**Pull request by kyle-sexton**
*This was generated by AI during work-loop execution.*

## Summary

- **Adds `check-listing-budget.sh` + a `listing-budget` action** — the shared/aggregate skill-listing
  character budget (`skillListingBudgetFraction`, default 1% of the model's context window) had no
  check at all. `check-skill.sh` check 2 only ever guarded the **per-skill** entry cap
  (`skillListingMaxDescChars`, 1536 chars) — a different, narrower limit. The new script pools one or
  more skills roots into a single aggregate estimate against a documented, overridable default (8000
  chars — the harness's own `SLASH_COMMAND_TOOL_CHAR_BUDGET` fallback), reports the biggest
  contributors on overflow, and is always advisory (exit 0): the live budget depends on a model's
  context window and a consumer's own settings, neither of which a static check can observe. Wired
  into the `skill-quality-gate` CI job as a report-only step pooling every plugin's `skills/` root into
  one marketplace-wide aggregate (currently ~109,000 chars against the 8,000-char default floor).
- **Fixes check 2's joiner (#1404 item 2).** The harness assembles a listing entry as `description` +
  `" - "` + `when_to_use` — a literal 3-char joiner — that check 2's sum omitted, under-counting by 3
  whenever `when_to_use` is populated. Not currently binding at present description lengths in this
  repo, but wrong in exactly the direction the listing-budget work is about.
- **Settles the issue's stated 4x discrepancy (#1404, "one discrepancy worth resolving").** Fetched
  current docs (`code.claude.com/docs/en/settings`, live HTML) confirm
  `SLASH_COMMAND_TOOL_CHAR_BUDGET`'s documented 8000-char fallback is exactly `contextTokens(200000) x
  ~4 chars/token x skillListingBudgetFraction(0.01)` — the `contextTokens x 4 x fraction` derivation
  from the shipped binary, not the bare `tokens x fraction` reading. `skillListingBudgetFraction`
  (default `0.01`) and `skillListingMaxDescChars` (default `1536`, matching this repo's existing
  `DESC_CHAR_CAP`) were also confirmed directly from the live settings page.
- **Item 4's caveat (never hardcode `skillListingBudgetFraction`'s default as a resolved live value)**
  is folded into the new script's design: every tunable (budget chars, context tokens, fraction,
  chars/token, per-entry cap) is environment-overridable, and the default budget is anchored on the
  one documented number (8000) rather than reconstructed from assumed inputs.

### Deferred: item 3 (check 12 amendment) — split into #1451

Item 3 — amending check 12 to accept a populated, single-quoted `when_to_use` as satisfying the
trigger-spec requirement instead of requiring a literal `Use when:` prefix — is **not** in this PR.
The issue itself sequences it behind #1096 ("already editing that file and claims check 21"), and I
verified #1096 is a real, currently open, active PR (branch `feat/fresh-eyes-delegation-doctrine-gate`,
708 additions) that does edit `check-skill.sh` (appending check 21). Per that sequencing note, item 3
is split out into a dedicated follow-up, **#1451**, scoped and ready to pick up once #1096 merges or is
otherwise resolved. This PR closes #1404 (items 1, 2, and 4 are complete; item 3's tracking now lives
in #1451, not in a reopened #1404).

## Test plan

- [x] New `check-listing-budget.test.sh` — 8 assertions (help, missing-root error, empty root,
      under-budget OK, forced-tiny-budget WARN, joiner counted, per-entry cap applied before summing,
      multi-root pooling). All pass.
- [x] `check-skill.test.sh` — added assertion 29 (boundary case: `desc(1500)+wtu(34)=1534` passes
      check 2 without the joiner fix, fails at `1537/1536` with it). Full suite: 29/29 pass.
- [x] `scripts/check-changed-skills.test.sh` (CI orchestrator self-test): 8/8 pass.
- [x] `scripts/check-skill-portability.test.sh`: 17/17 pass; `check-skill-portability.sh origin/main`
      reports no unexcused coupling tokens in the changed skill file.
- [x] Dogfooded `check-skill.sh` against the edited `skill-quality:check` skill itself: PASS, 0
      errors, 0 warnings, all 7 base-ref trigger phrases preserved (description rewrite kept every
      existing single-quoted trigger verbatim and only added new ones).
- [x] `check-changed-skills.sh origin/main` (the actual CI gate step): PASS.
- [x] `shellcheck --rcfile=.shellcheckrc` clean on both new scripts (fixed one SC2016 literal-backtick
      note and one SC2034 unused-variable warning during development).
- [x] `actionlint .github/workflows/ci.yml`: clean.
- [x] `markdownlint-cli2` clean on the three edited Markdown files.
- [x] Manually ran `check-listing-budget.sh plugins/*/skills` against this repo: 183 skills, 61 roots,
      aggregate 108,975 chars vs the 8,000-char default budget (~13.6x over) — reproduces the
      overflow the issue's own measurement found, with concrete top contributors.

## Related

Closes #1404 (items 1, 2, and 4 from that issue; item 3 is split into #1451, not resolved here).
Refs #1451 (new follow-up: item 3, check 12 amendment — sequenced behind #1096, ready to pick up once
it merges or is otherwise resolved).
Refs #1096 (open PR already editing `check-skill.sh`; #1451 sequences behind it).
Refs #1271 (the original listing-budget issue #1404 was extracted from).

### CODE REVIEW

**Comment by kyle-sexton** [comment]
*This was generated by AI during work-loop execution.*

Addressed the automated review's minor finding (`--help` sed range clipping the header's last line) in b2d3f7f8 — `2,79p` -> `2,80p`. Verified `--help` output now includes the final header line and re-ran the full test suite + shellcheck (clean).

Security review found no issues; no other findings to address.

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  [[ -d "$root" ]] || continue
+  found_any_root=1
+  for skill_md in "$root"/*/SKILL.md; do
+    [[ -f "$skill_md" ]] || continue
+    skill_name="${skill_md%/SKILL.md}"
+    skill_name="${skill_name##*/}"
+    fm="$(skill_frontmatter::extract <"$skill_md")"
+    [[ -n "$fm" ]] || continue
+    desc="$(skill_frontmatter::strip_quotes "$(skill_frontmatter::field description <<<"$fm")")"
+    wtu="$(skill_frontmatter::strip_quotes "$(skill_frontmatter::field when_to_use <<<"$fm")")"
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh:262]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 1 | Exclude non-model-invocable skills from the aggregate | **VALID — fixed in d218d15d** | Confirmed against current official docs fetched this session, per this repo's fresh-docs mandate | 👍 |

Correct, and the harness premise checks out. From <https://code.claude.com/docs/en/skills>: the invocation-control table records **"Description not in context"** for `disable-model-invocation: true`, and *Hide individual skills* states it **"removes the skill from Claude's context entirely."** The shared budget is spent on description + `when_to_use` text, so those skills spend none of it and counting them overstated the report.

Your count is exact: **51 of 183** `SKILL.md` files under `plugins/*/skills` set the flag. With the filter, the CI report moves from **183 skills / 109,205 chars** to **132 / 83,594** — still ~10.4× over the 8000-char default, so the finding this check exists to surface is unchanged.

Documented alongside the filter in the script header, `SKILL.md`, `README.md`, and the changelog, with one caveat worth recording: a consumer's `skillOverrides` can free further descriptions via `"name-only"`, which repository content cannot reveal — so the reported figure is an *upper* bound for anyone who sets it. Test 9 pins the behaviour (`dmi: true` excluded, an explicit `dmi: false` still counted).

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+TOTAL=0
+ENTRY_COUNT=0
+CONTRIB_FILE="$(mktemp)"
+trap 'rm -f "$CONTRIB_FILE"' EXIT
+
+found_any_root=0
+for root in "${ROOTS[@]}"; do
+  [[ -d "$root" ]] || continue
+  found_any_root=1
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 2 | Reject every missing explicit skills root | **VALID — fixed in d218d15d** | Reproduced on the branch, then fixed and pinned by tests | 👍 |

Reproduced exactly as described: `check-listing-budget.sh <valid-root> <missing-root>` printed `across 2 root(s)`, scanned only one, and exited 0. `found_any_root` tracked "at least one existed", so only the all-missing case was guarded.

Both halves fixed. An explicit root that does not exist is now an environment error — `Error: skills root does not exist: <root>`, exit 2 — while the no-args resolution path keeps its distinct `no skills root found` message, so the two branches stay separable. The header now reports roots **actually scanned** (`FOUND_ROOTS`) instead of `${#ROOTS[@]}`.

Tests 2, 2b, and 2c pin the missing-explicit-root, partial-missing (one valid + one missing), and no-args-resolution cases respectively. The documented CI form `check-listing-budget.sh plugins/*/skills` still exits 0 — verified by running it, not inferred: glob expansion only yields existing paths.

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+MAX_DESC_CHARS="${CHECK_SKILL_LISTING_MAX_DESC_CHARS:-1536}"
+JOINER_CHARS=3 # the literal " - " the harness inserts between description and when_to_use
+
+if [[ -n "${CHECK_SKILL_LISTING_CONTEXT_TOKENS:-}" ]]; then
+  CONTEXT_TOKENS="$CHECK_SKILL_LISTING_CONTEXT_TOKENS"
+  CHARS_PER_TOKEN="${CHECK_SKILL_LISTING_CHARS_PER_TOKEN:-4}"
+  FRACTION="${CHECK_SKILL_LISTING_BUDGET_FRACTION:-0.01}"
+  if ! BUDGET_CHARS="$(awk -v t="$CONTEXT_TOKENS" -v c="$CHARS_PER_TOKEN" -v f="$FRACTION" \
+    'BEGIN { printf "%d", t * c * f }' 2>/dev/null)" || [[ -z "$BUDGET_CHARS" ]]; then
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 3 | Validate numeric overrides before computing the budget | **VALID — fixed in d218d15d** | Both failure modes reproduced on the branch | 👍 |

Both halves reproduced. `CHECK_SKILL_LISTING_CONTEXT_TOKENS=nope` printed `budget: 0 chars (reconstructed: nope tokens x 4 chars/token x 0.01)` with a WARN and **exit 0** — awk exits 0 and prints `0` for a nonnumeric operand, so neither the `||` nor the `-z` guard tripped and the error branch was dead. `CHECK_SKILL_LISTING_MAX_DESC_CHARS=nope` and `CHECK_SKILL_LISTING_BUDGET_CHARS=nope` instead died with `unbound variable` / `invalid number` at **exit 1**, which the header never documented.

All five numeric overrides are now validated as positive numbers before use, with failures routed into the already-documented environment error (exit 2). The validator distinguishes integer counts from the decimal ratio/fraction, so the documented `0.01` default shape is still accepted — an integer-only guard would have rejected valid input. Undocumented exit 1 is eliminated; the budget verdict itself stays advisory (OK and WARN both exit 0).

Tests: six rejection cases (each asserting the variable actually blamed) plus one accepting `CHARS_PER_TOKEN=3.5` with `BUDGET_FRACTION=0.02`.

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    printf 'Error: could not compute a budget from CHECK_SKILL_LISTING_CONTEXT_TOKENS=%s CHECK_SKILL_LISTING_CHARS_PER_TOKEN=%s CHECK_SKILL_LISTING_BUDGET_FRACTION=%s\n' \
+      "$CONTEXT_TOKENS" "$CHARS_PER_TOKEN" "$FRACTION" >&2
+    exit 2
+  fi
+  BUDGET_SOURCE="reconstructed: $CONTEXT_TOKENS tokens x $CHARS_PER_TOKEN chars/token x $FRACTION"
+else
+  BUDGET_CHARS="${CHECK_SKILL_LISTING_BUDGET_CHARS:-8000}"
+  BUDGET_SOURCE="documented default (SLASH_COMMAND_TOOL_CHAR_BUDGET fallback)"
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 4 | Identify fixed budget values as overrides | **VALID — fixed in d218d15d** | Reproduced on the branch; both arms now asserted | 👍 |

Reproduced: `CHECK_SKILL_LISTING_BUDGET_CHARS=10` printed `budget: 10 chars (documented default (SLASH_COMMAND_TOOL_CHAR_BUDGET fallback))`. The value and the label were assigned unconditionally in the same arm, so nothing distinguished the `:-8000` fallback from a supplied value.

A supplied value is now labelled `override (CHECK_SKILL_LISTING_BUDGET_CHARS)`; only the unset case keeps the documented-default label. That restores the provenance distinction the action's own documentation promises.

Tests 12 and 12b pin both arms. Worth noting why this shipped green: the previous suite *did* set this variable, but asserted only the WARN path and never the label — so the mislabel was invisible to it.

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+  exit 2
+fi
+
+# shellcheck source=./skill-frontmatter.sh
+source "$SCRIPT_DIR/skill-frontmatter.sh"
+
+MAX_DESC_CHARS="${CHECK_SKILL_LISTING_MAX_DESC_CHARS:-1536}"
+JOINER_CHARS=3 # the literal " - " the harness inserts between description and when_to_use
+
+if [[ -n "${CHECK_SKILL_LISTING_CONTEXT_TOKENS:-}" ]]; then
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 5 | Prioritize the fixed budget override | **VALID — fixed in d218d15d** | Reproduced on the branch; verdict flip confirmed possible | 👍 |

Reproduced: with `CHECK_SKILL_LISTING_BUDGET_CHARS=99999 CHECK_SKILL_LISTING_CONTEXT_TOKENS=200000` the report printed `budget: 8000 chars (reconstructed: ...)` and the supplied 99,999 appeared nowhere in the output. The verdict flip is real rather than hypothetical — at an aggregate of ~50,000 the fixed budget yields OK while the reconstruction yields WARN.

This contradicted the script's own header verbatim, which has always said `CHECK_SKILL_LISTING_BUDGET_CHARS` "skips the token/fraction reconstruction below". The fixed override is now checked first, so that sentence is true, and the ignored reconstruction input is **announced** on stderr rather than dropped silently.

Of your two suggested remedies I took the reorder rather than rejecting the conflicting inputs: the header already declares the precedence, so honouring it is the smaller change and adds no new failure mode for anyone currently setting both. Test 13 pins the precedence and the notice together.

**Comment by kyle-sexton** [comment]
*This was generated by AI during work-loop execution.*

All open review findings are addressed in d218d15d — the three the re-review listed as remaining, plus the two further Codex P2s. Per-finding evidence is in each inline thread; summary here since the last top-level comment still reads "three items remain before merge".

| # | Finding | Source | Classification |
|---|---------|--------|----------------|
| 1 | `disable-model-invocation: true` skills included in the aggregate | Codex P1 / re-review 3 | **VALID — fixed** |
| 2 | Missing explicit skills root silently skipped; header counts args, not scanned roots | Codex P2 / re-review 1 | **VALID — fixed** |
| 3 | Nonnumeric overrides coerced to zero (or exit 1) instead of the documented exit 2 | Codex P2 | **VALID — fixed** |
| 4 | `CHECK_SKILL_LISTING_BUDGET_CHARS` mislabelled "documented default" | Codex P2 / re-review 2 | **VALID — fixed** |
| 5 | Fixed budget silently ignored when `CHECK_SKILL_LISTING_CONTEXT_TOKENS` is also set | Codex P2 | **VALID — fixed** |

**On the P1 design question**, which the prior review flagged as "worth confirming before committing" — confirmed against current official documentation rather than recall, per this repo's fresh-docs mandate. <https://code.claude.com/docs/en/skills>: the invocation-control table records **"Description not in context"** for `disable-model-invocation: true`, and *Hide individual skills* states it **"removes the skill from Claude's context entirely."** Those skills spend none of the shared description budget, so the filter is a correctness fix. The CI report moves from **183 skills / 109,205 chars** to **132 / 83,594** — still ~10.4× over the 8000-char default, so the finding this check exists to surface is unchanged. The confirmation is recorded in the script header, `SKILL.md`, `README.md`, and the changelog so a future maintainer does not have to re-litigate it.

**Semantics deliberately unchanged.** The budget verdict stays advisory: OK *and* WARN both exit 0. The new hard failures use exit 2, which was already the script's documented usage/env-error class — and one of them *removes* an undocumented exit 1 that a nonnumeric override could previously trigger. Nothing in this commit touches `check-skill.sh`'s check-2 PASS/FAIL surface.

**Verification on d218d15d:** `check-listing-budget.test.sh` 21/21 (was 8 — findings 2, 3 and 4 had all shipped green precisely because no assertion covered them); `check-skill.test.sh` 7/7; `shellcheck -x` clean on both scripts; `markdownlint-cli2` clean on the three changed docs; the skill-quality gate self-check PASS with all 9 base-ref trigger phrases preserved and the description at 972/1536; and the literal CI invocation `check-listing-budget.sh plugins/*/skills` exits 0.

File: plugins/skill-quality/scripts/check-listing-budget.sh [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    [[ -f "$skill_md" ]] || continue
+    skill_name="${skill_md%/SKILL.md}"
+    skill_name="${skill_name##*/}"
+    fm="$(skill_frontmatter::extract <"$skill_md")"
+    [[ -n "$fm" ]] || continue
+    # `disable-model-invocation: true` keeps this skill's description out of
+    # the model-visible listing entirely, so it spends none of the shared
+    # budget — counting it would overstate the aggregate. See the header.
+    dmi="$(skill_frontmatter::strip_quotes "$(skill_frontmatter::field disable-model-invocation <<<"$fm")")"
+    [[ "$dmi" == "true" ]] && continue
```
[the remark this answers is not in the document]
Comment (kyle-sexton) [reply plugins/skill-quality/scripts/check-listing-budget.sh]: | # | Finding | Classification | Evidence | Reacted |
|---|---------|---------------|----------|---------|
| 6 | Normalize the invocation-control boolean before filtering | **VALID — fixed in f4121042** | Reproduced against the updated script before fixing | 👍 |

Confirmed, and worse than the single case you cited. Built a nine-skill fixture root and ran the updated script against it before changing anything: it reported **7 listing-eligible skills instead of 2**. Five spellings leaked through the `== "true"` compare — `true # manual-only`, `'true'  # manual`, `TRUE`, `True`, and a trailing-whitespace value. Only the bare `"true"` / `'true'` forms were caught, and only because `strip_quotes` happened to handle them.

Fixed by normalizing before comparison: strip a whitespace-preceded YAML comment, trim surrounding whitespace, strip one quote layer, fold ASCII case.

**Scope bounded deliberately, and recorded in the code so it does not get widened by accident:**

- Comment-stripping applies **only** to this boolean — never to `description` or `when_to_use`, where a whitespace-preceded `#` is content. Folding it there would silently truncate a description, trading this bug for a worse one.
- YAML 1.1's `yes` / `on` aliases are **not** folded. The documented spelling is `true`; over-matching risks dropping a skill over a value the harness may read as a plain string. Flagging that as a conscious call rather than an oversight.
- It is a pragmatic normalizer for one known field, not a YAML parser.

**Checked for the same class of gap elsewhere**, since one push is cheaper than a third round: this was the only boolean compared against raw field output anywhere in the plugin. Every other `skill_frontmatter::field` consumer extracts free text, and `skill_frontmatter::metadata_field` already strips trailing comments internally — so the shared library needed no change, which keeps the blast radius off `check-skill.sh`'s PASS/FAIL surface.

**One correction this surfaced.** The marketplace figure is **132 skills / 83,611 chars**, not the 83,594 recorded earlier — that measurement predated this PR's own `SKILL.md` description edit, which added 17 characters. The changelog now carries the figure with the commit it was measured at, per this repo's own convention that a figure without its commit goes stale.

**Verification on f4121042:** `check-listing-budget.test.sh` 22/22 (the new case fails against the pre-fix code — verified, not assumed); `check-skill.test.sh` 7/7; `shellcheck -x` clean; `markdownlint-cli2` clean; skill-quality gate PASS with all 9 base-ref trigger phrases preserved; `check-listing-budget.sh plugins/*/skills` exits 0.
